lib/configuration.js

```diff
@@ -15,150 +15,615 @@
+};
+/**
+ * Returns the _key property on the instance.
+ * @memberof Configuration
+ * @public
+ * @function getKey
+ * @returns {String|Null} - returns the _key property
+ */
+Configuration.prototype.getKey = function() {
+  return this._key;
+};
+/**
+ * Returns the _serviceContext property on the instance.
+ * @memberof Configuration
+ * @public
+ * @function getKey
+ * @returns {Object|Null} - returns the _serviceContext property
+ */
+Configuration.prototype.getServiceContext = function() {
+  return this._serviceContext;
+};
+/**
+ * Returns the _version property on the instance.
+ * @memberof Configuration
+ * @public
+ * @function getVersion
+ * @returns {String} - returns the _version property
+ */
+Configuration.prototype.getVersion = function() {
+  return this._version;
+};
+/**
+ * Returns the _initError property on the instance.
+ * @memberof Configuration
+ * @public
+ * @function getError
+ * @returns {Error|Null} - returns the _initError property
+ */
+Configuration.prototype.getError = function() {
+  return this._initError;
+};
+/**
+ * Returns a Boolean based upon whethe the Configuration instance communicated
+ * with the metadata service and has a sufficient configuration.
+ * @memberof Configuration
+ * @public
+ * @function isReady
+ * @returns {Boolean} - returns true if instance initialized without error,
+ *  otherwise false
+ */
+Configuration.prototype.isReady = function() {
+  return this._initState && !this._initError;
+};
+/**
+ * Returns a Boolean indicating whether or not the Configuration instance went
+ * through the initialization process and encountered an error during that
+ * process.
+ * @memberof Configuration
+ * @public
+ * @function hasErrored
+ * @returns {Boolean} - returns true if _initError is set as an Error, otherwise
+ *  false
+ */
+Configuration.prototype.hasErrored = function() {
+  return this._initError instanceof Error;
+};
+/**
+ * The init function causes the Configuration instance to being the
+ * initialization process of gathering configuration from both the metadata
+ * service, runtime configuration and environment. This function will begin the
+ * init process only once. The function will defer starting the initialization
+ * process on any successive calls.
+ * @memberof Configuration
+ * @public
+ * @function init
+ * @chainable
+ * @returns {this} - returns the instance for chaining
+ */
+Configuration.prototype.init = function () {
+  if (this._startedInit === false) {
+    /**
+     * Attempt to get the project number from the metadata service
+     * and kick-off the init process. This would be where we kick off the
+     * `getProjectId` invocation but since that version is not live yet we use
+     * project number for now.
+     * @todo (cristiancavalli) - turn this into the getProjectId call when it
+     *  goes live and remove the getProjectNumber function and call.
+     * @example
+     * utils.getProjectId(this._assimilateProjectId.bind(this));
```

Comment:
It does not, it was a derivation of the function name `_assimilateProjectNumber`.
I was planning on using that schema `_assimilateProject{id-type}` for the config data sink and `_assimilateProjectId` when the common diag module is published with the new getProjectId functionality